src/lib/support/ScopedBuffer.h

```diff
@@ -177,28 +177,31 @@ class ScopedMemoryBufferWithSize : public ScopedMemoryBuffer<T>
     {
         if (this != &other)
         {
-            mSize       = other.mSize;
-            other.mSize = 0;
+            mCount       = other.mCount;
+            other.mCount = 0;
         }
         ScopedMemoryBuffer<T>::operator=(std::move(other));
         return *this;
     }
 
-    ~ScopedMemoryBufferWithSize() { mSize = 0; }
+    ~ScopedMemoryBufferWithSize() { mCount = 0; }
 
     // return the size in bytes
-    inline size_t AllocatedSize() const { return mSize; }
+    inline size_t AllocatedSize() const { return mCount * sizeof(T); }
```

Comment:
size_t is used to count objects and doesn't need to mean 'bytes'.  Likewise, size() in STL containers also refers to the number of contained objects, not the number of bytes. This class is following that pattern.

Do we need to change the class then?  Obviously this comment was incorrect: `// return the size in bytes`.  But couldn't we just leave this class as is and fix this comment instead?  This seems like unnecessary churn.

Reply:
Even though the previously incorrect comment, `// return the size in bytes`, claimed we were returning bytes, we weren't.  So no code currently needs such an API and I don't particularly see the usefulness for it in the future.

Given that, instead of the changes in this PR, one could literally just update the comment on 189 to reflect what this API actually previously returned, which was a count of objects.

Regardless of the naming, I don't see that we actually need to have a separate bytes-returning-api.  So if this new naming is preferred, you might consider removing the now-unused `AllocatedSize` interface.

Reply:
Actually, it was indeed returning count of bytes before my change here.... I think the original code suffered some ambiguity in that `ScopedMemoryBuffer::Calloc` and `ScopedMemoryBuffer::Alloc` had different argument names (elementCount vs size), but treated them the same - as a "element count".

In any case, sounds like we should just pick one and use it for "count of objects", and I'm okay with using "size" for it - I'll remove `AllocatedCount` and fix the comment for `AllocatedSize`.